numpy/_core/src/multiarray/array_assign_scalar.c

```diff
@@ -86,13 +86,19 @@ raw_array_assign_scalar(int ndim, npy_intp const *shape,
         NPY_BEGIN_THREADS_THRESHOLDED(nitems);
     }
 
+    if (((int)casting & NPY_SAME_VALUE_CASTING_FLAG) > 0) {
```

Comment:
Would it not be simpler to just pass on `casting` itself inside `cast_info.context` instead of new flags?

Reply:
FWIW, I don't mind either way.  Adding flags makes sense either way, but passing on casting is also fine.
I suspect part of the reason is that right now many parts don't set this correctly yet and if there was a `casting` then that might be more confusing (compared to a flag).

OTOH, as this is the only use for a casting field maybe it doesn't matter and we should do what we think is simpler.